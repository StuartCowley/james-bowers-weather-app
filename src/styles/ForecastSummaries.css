.forecast-summaries {
    font-family: 'Work Sans', sans-serif;
    display: flex;
    flex-direction: row;
    /* flex-flow: row wrap; */
    justify-content: space-between;
    align-items: center;
    text-align: center;
    color: white;
    text-shadow: 0 0 5px rgba(0, 0, 0, 0.5);

}
.forecast-summary{
    outline: 2px solid white;
    padding: 8px;
    margin: 6px;
    background-color: rgba(255, 255, 255, 0.4);
    -moz-box-shadow:    inset 0 0 20px rgba(0, 0, 0, 0.2);
    -webkit-box-shadow: inset 0 0 20px rgba(0, 0, 0, 0.2);
    box-shadow:         inset 0 0 20px rgba(0, 0, 0, 0.2),
                              -10px 10px 20px rgba(0,0,0,0.2);
    /* min-width: 180px; */
}
.forecast-summary__date{
    font-size: 1.5rem;
    font-weight: 400;
    color: white;
}
.forecast-summary__icon {
    display: flex;
    align-items: center;
    justify-content: center;
    font-size: 4.5rem;
    height: 10rem;
    color: white;
    text-shadow: 0 0 20px rgba(0, 0, 0, 0.5);
}
.forecast-summary__maxTemp {
    margin: 5px;
    font-weight: 300;
    font-size: 1.5rem;
}
.forecast-summary__description {
    font-weight: 600;
    font-size: 1.5rem;
}