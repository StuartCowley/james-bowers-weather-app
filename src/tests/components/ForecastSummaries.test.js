import React from "react";
import { render } from "@testing-library/react";
import ForecastSummaries from "../../components/ForecastSummaries";

describe("ForecastSummaries", () => {
  const validProps = {
    forecasts: [
      {
        date: 1111111111111,
        description: "Stub description",
        icon: "800",
        temperature: {
          max: 22,
          min: 12,
        },
      },
      {
        date: 2222222222222,
        description: "Stub description2",
        icon: "211",
        temperature: {
          max: 17,
          min: 10,
        },
      },
      {
        date: 3333333333333,
        description: "Stub description3",
        icon: "602",
        temperature: {
          max: 18,
          min: 11,
        },
      },
      {
        date: 4444444444444,
        description: "Stub description4",
        icon: "781",
        temperature: {
          max: 19,
          min: 12,
        },
      },
      {
        date: 5555555555555,
        description: "Stub description5",
        icon: "721",
        temperature: {
          max: 24,
          min: 13,
        },
      },
    ],
    onForecastSelect: () => {},
  };

  it("renders correctly", () => {
    const { asFragment } = render(
      <ForecastSummaries
        forecasts={validProps.forecasts}
        onForecastSelect={validProps.onForecastSelect}
      />
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it("renders the correct amount of summaries", () => {
    const { getAllByTestId } = render(
      <ForecastSummaries
        forecasts={validProps.forecasts}
        onForecastSelect={validProps.onForecastSelect}
      />
    );

    expect(getAllByTestId("forecast-summary")).toHaveLength(
      validProps.forecasts.length
    );
  });
});
