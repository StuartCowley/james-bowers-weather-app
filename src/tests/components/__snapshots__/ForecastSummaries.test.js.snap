// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForecastSummaries renders correctly 1`] = `
<DocumentFragment>
  <div
    class="forecast-summaries"
  >
    <div
      class="forecast-summary"
      data-testid="forecast-summary"
    >
      <div
        class="forecast-summary__date"
      >
        Fri Mar 18
      </div>
      <div
        class="forecast-summary__icon"
        data-testid="forecast-icon"
      >
        <i
          class="wi wi-day-sunny"
          name="owm"
        />
      </div>
      <div
        class="forecast-summary__maxTemp"
      >
        <p
          class="forecast-summary__maxTemp"
        >
          22°C
        </p>
      </div>
      <div
        class="forecast-summary__description"
      >
        Stub description
      </div>
      <button
        class="forecast-summary__more-details"
        type="button"
      >
        More details
      </button>
    </div>
    <div
      class="forecast-summary"
      data-testid="forecast-summary"
    >
      <div
        class="forecast-summary__date"
      >
        Sat Jun 02
      </div>
      <div
        class="forecast-summary__icon"
        data-testid="forecast-icon"
      >
        <i
          class="wi wi-day-thunderstorm"
          name="owm"
        />
      </div>
      <div
        class="forecast-summary__maxTemp"
      >
        <p
          class="forecast-summary__maxTemp"
        >
          17°C
        </p>
      </div>
      <div
        class="forecast-summary__description"
      >
        Stub description2
      </div>
      <button
        class="forecast-summary__more-details"
        type="button"
      >
        More details
      </button>
    </div>
    <div
      class="forecast-summary"
      data-testid="forecast-summary"
    >
      <div
        class="forecast-summary__date"
      >
        Sun Aug 18
      </div>
      <div
        class="forecast-summary__icon"
        data-testid="forecast-icon"
      >
        <i
          class="wi wi-day-snow"
          name="owm"
        />
      </div>
      <div
        class="forecast-summary__maxTemp"
      >
        <p
          class="forecast-summary__maxTemp"
        >
          18°C
        </p>
      </div>
      <div
        class="forecast-summary__description"
      >
        Stub description3
      </div>
      <button
        class="forecast-summary__more-details"
        type="button"
      >
        More details
      </button>
    </div>
    <div
      class="forecast-summary"
      data-testid="forecast-summary"
    >
      <div
        class="forecast-summary__date"
      >
        Mon Nov 03
      </div>
      <div
        class="forecast-summary__icon"
        data-testid="forecast-icon"
      >
        <i
          class="wi wi-tornado"
          name="owm"
        />
      </div>
      <div
        class="forecast-summary__maxTemp"
      >
        <p
          class="forecast-summary__maxTemp"
        >
          19°C
        </p>
      </div>
      <div
        class="forecast-summary__description"
      >
        Stub description4
      </div>
      <button
        class="forecast-summary__more-details"
        type="button"
      >
        More details
      </button>
    </div>
    <div
      class="forecast-summary"
      data-testid="forecast-summary"
    >
      <div
        class="forecast-summary__date"
      >
        Tue Jan 18
      </div>
      <div
        class="forecast-summary__icon"
        data-testid="forecast-icon"
      >
        <i
          class="wi wi-day-haze"
          name="owm"
        />
      </div>
      <div
        class="forecast-summary__maxTemp"
      >
        <p
          class="forecast-summary__maxTemp"
        >
          24°C
        </p>
      </div>
      <div
        class="forecast-summary__description"
      >
        Stub description5
      </div>
      <button
        class="forecast-summary__more-details"
        type="button"
      >
        More details
      </button>
    </div>
  </div>
</DocumentFragment>
`;
